spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/MySqlJdbcInfoProviderAutoConfiguration.java

```diff
@@ -35,6 +36,7 @@
 		com.mysql.jdbc.Driver.class})
 @ConditionalOnProperty(
 		name = "spring.cloud.gcp.sql.enabled", havingValue = "true", matchIfMissing = true)
+@AutoConfigureAfter(AppEngineJdbcInfoProviderAutoConfiguration.class)
```

Comment:
Why still no `@AutoConfigureBefore(GcpCloudSqlAutoConfiguration.class)` ?

Reply:
Isn't that taken care of naturally by the dependency graph?

i.e., `GcpCloudSqlAutoConfiguration` requires a `CloudSqlJdbcInfoProvider`, so the other auto configs will be tried first?

Reply:
Oh! I don't know... There is no any hints on the `@AutoConfigureAfter/@AutoConfigureBefore` to indicate such a situation. But, yeah. You might be right, that import for the `CloudSqlJdbcInfoProvider` should trigger those auto-configs.

We may ask @snicoll to enlighten this subject.
Thanks

Reply:
Tests are passing and I'm fairly certain that the dependency graph is solving the issue for us.
So I would suggest that if @snicoll detects any issue when he comes back tomorrow, we issue another small PR to fix?

Reply:
> GcpCloudSqlAutoConfiguration requires a CloudSqlJdbcInfoProvider, so the other auto configs will be tried first?

As long as you are injecting a bean by type (and only that) what @joaoandremartins said. But as soon as you are playing with conditions, it will break.

My advice would be to order auto-configurations properly regardless.

Reply:
OK, added `@AutoConfigureBefore(GcpCloudSqlAutoConfiguration.class)` to `AppEngineJdbcInfoProviderAutoConfiguration`.
Together with the `@AutoConfigureAfter(AppEngineJdbcInfoProviderAutoConfiguration.class)`'s in `MySqlJdbcInfoProviderAutoConfiguration` and `PostgreSqlJdbcInfoProviderAutoConfiguration`, that's hopefully enough ordering.